File: src/main/java/de/akquinet/ccsp/floodmark/datamodel/MeasurementImport.java
package de.akquinet.ccsp.floodmark.datamodel;

import java.util.Date;
import lombok.Data;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
public class MeasurementImport {

    private String id;

    private final String source;

    @Indexed(unique = true)
    private final Date timeOfCreation;
}

File: src/main/kotlin/de/akquinet/ccsp/floodwatch/datamodel/MeasurementImport.kt
package de.akquinet.ccsp.floodwatch.datamodel

import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document
data class MeasurementImport(val source: String,
                             @field:Indexed val timeOfCreation: Date) {
    var id: String? = null
}
